src/cmd/pay.rs

```diff
@@ -259,15 +277,71 @@ pub enum TokenInput {
     Hst,
 }
 
-// By default, helium-api-rs serializes and deserializes tokens as u64s (ie: bones).
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(untagged)]
+enum Amount {
+    Dec(Decimal),
+    Str(String),
```

Comment:
Is it possible to make `Str(String)` case simply `Max`?

Reply:
Heh. I had it that way until last night, when I "simplified" things. Here are the options as I see them:

1) Amount::Max
Amount is 
```enum Amount {
    Value(Decimal),
    Max,
}```

This feels right, but I couldn't get the custom deserializer to work. Based on some reading online, I expected the custom deserializer to deserialize a string, then I could test for `max` or do `Decimal.from_str()`, returning errors if neither of those worked. This didn't work for me, despite what StackOverflow said. Numeric input in the multi JSON like `"amount": 1.23` resulted in an error when I did: `let raw_string: String = Deserialize::deserialize(deserializer)?;` I tried several different ways, including trying to deserialize as an f64 first, then converting that to Decimal, but that also didn't work. I don't know enough about serde Deserializer to make it work, and I couldn't learn enough in the time I had. So, I switched to option two.

2) Take advantage of serde's `untagged` feature. 
Amount is the same as above, but I also add an intermediate enum called StrOrDec.
```
#[serde(untagged)]
enum StrOrDec {
    Dec(Decimal),
    Str(String),
}```

Then in the custom deserializer I can do `let raw: StrOrDec = Deserialize::deserialize(deserializer)?;`. Serde will automatically try to deserialize first into a Decimal and if that fails into a String, giving me the expected variant of StrOrDec. From there the custom deserializer creates the Amount object and returns it. This is what I had up until last night. But it felt "ugly" to me, so I moved on to the third option, which is where the PR is as of 2020-09-22 22:00:00 PDT.

3) Amount is the untagged enum, and it has accessor functions that contain all the logic around determining max, returning a valid amount or zero.
```
#[serde(untagged)]
enum Amount {
    Dec(Decimal),
    Str(String),

    pub fn token_amount()
    pub fn is_max()
    pub fn is valid()
}```

This option lets me keep all the details about Amount (return 0 amount  when "max" is true, what a valid or invalid Amount object is, etc ) inside Amount. The `collect_payments()` function remains very simple. The custom deserializer is simple. This feels cleaner to me.

IMO, the best option would be a combination of 1 and 3. Use Amount::Max and Amount::Value(Decimal), figure out how to get the customer deserializer to do this, but also keep Amount::is_max() and Amount::token_amount() to keep the details of Amount inside. But, I could not figure it out. If someone has advice, sample code, or just links how to get a custom deserializer try to deserialize a String or str, fail gracefully, then deserialize a Decimal, I would happily do that.

Reply:
Ok, I took another pass at this and have it working! Amount now has "Dec" and "Max" variants, and handles the "amount is 0 is Amount::Max" logic.